src/Aspect/ContractCheckerAspect.php

```diff
@@ -52,21 +49,7 @@ public function __construct(Reader $reader)
      */
     public function preConditionContract(MethodInvocation $invocation)
     {
-        $object = $invocation->getThis();
-        $args   = $this->getMethodArguments($invocation);
-        $scope  = $invocation->getMethod()->getDeclaringClass()->name;
-
-        foreach ($invocation->getMethod()->getAnnotations() as $annotation) {
-            if (!$annotation instanceof Contract\Verify) {
-                continue;
-            }
-
-            try {
-                $this->ensureContractSatisfied($object, $scope, $args, $annotation);
-            } catch (\Exception $e) {
-                throw new ContractViolation($invocation, $annotation->value, $e);
-            }
-        }
+        (new PreconditionContract($this->reader))->check($invocation);
```

Comment:
All right, we can move object initialization to ContractCheckerAspect::construct